client.go

```diff
@@ -30,8 +31,9 @@ type Client struct {
 	Net            string            // if "tcp" or "tcp-tls" (DNS over TLS) a TCP query will be initiated, otherwise an UDP one (default is "" for UDP)
 	UDPSize        uint16            // minimum receive buffer for UDP messages
 	TLSConfig      *tls.Config       // TLS connection configuration
-	Timeout        time.Duration     // a cumulative timeout for dial, write and read, defaults to 0 (disabled) - overrides DialTimeout, ReadTimeout and WriteTimeout when non-zero
-	DialTimeout    time.Duration     // net.DialTimeout, defaults to 2 seconds - overridden by Timeout when that value is non-zero
+	Dialer         *net.Dialer       // a net.Dialer used to set local address, timeouts and more
```

Comment:
Why is `net.Dailer` a pointer here, but in `Dial` it is used as-is `var d net.Dialer`

Reply:
it's a pointer so it can be `nil`, which means "I don't want to set any Dialer, use your defaults".

In `Dial` I do this:
```
    var d net.Dialer 
    if c.Dialer == nil {
        d = net.Dialer{}
    } else { 
        d = net.Dialer(*c.Dialer) 
    } 
```
which eventually uses another dialer instance - the one from `Client`, if any, or a default one if not